netkan/netkan/scheduler.py

```diff
@@ -57,26 +104,21 @@ def can_schedule(self, max_queued, dev=False, min_credits=200):
             )
             instance_id = response.text
             cloudwatch = boto3.client('cloudwatch')
-            stats = cloudwatch.get_metric_statistics(
-                Dimensions=[{'Name': 'InstanceId', 'Value': instance_id}],
-                MetricName='CPUCreditBalance',
-                Namespace='AWS/EC2',
-                StartTime=start.strftime("%Y-%m-%dT%H:%MZ"),
-                EndTime=end.strftime("%Y-%m-%dT%H:%MZ"),
-                Period=10,
-                Statistics=['Average'],
-            )
-            # A pass consumes around 40 credits, with an accrue rate of 24/hr.
-            # So running every 2 hours should see using just a touch less than
-            # we gain in that time period.
-            creds = 0
-            try:
-                creds = stats['Datapoints'][0]['Average']
-            except IndexError:
-                logging.error("Couldn't acquire CPU Credit Stats")
-            if int(creds) < min_credits:
+
+            cpu_credits = self.cpu_credits(cloudwatch, instance_id, start, end)
+            if cpu_credits < min_credits:
+                logging.info(
+                    "Run skipped, below cpu credit target (Current Avg: %s)", cpu_credits
+                )
+                return False
+
+            # Volume Burst balance measured in a percentage of 5.4million credits. Credits are
+            # accrued at a rate of 3 per GB, per second. If we are are down to 30 percent of
+            # our max, something has gone wrong and we should not queue any more inflations.
+            vol_credits = self.volume_credits(cloudwatch, instance_id, start, end)
+            if vol_credits < 30:
```

Comment:
The `min_credits` for the CPU are supplied via an argument. Should we consider the same for the volume credits? I know it's a percentage whereas the CPU credits are a total amount, so when we're upgrading the effect is not so severe, but it might still make sense.

Reply:
Maybe, but this is more of a fall safe, rather than something we expect to hit regularly. Maybe it should log error, so we know when this has happened.